src/newSeed/stage4.ts

```diff
@@ -76,10 +76,11 @@ export class Stage4 extends BaseStage {
     // Get the start and end time
     const startTimes = this.startTime.split(":");
     const endTimes = this.endTime.split(":");
-    let temp = this.startDate;
+    const temp:Date = undefined;
+    Object.assign(temp, this.startDate);
```

Comment:
I realized that assigning the temp to the start date changes the start time and end date object since the reference is only passed during the assignment. This change to the object affects the object and makes it difficult to appropriately check start and end times.